### ISSUE TITLE
`winget` mixes up PowerShell and PowerShell Preview

### ISSUE THREAD

**Issue by sba923** [microsoft/winget-cli#3798]
### Brief description of your issue

On a Windows 11 machine with both PowerShell 7.3.8.0 and PowerShell 7.4.0-preview.6 installed, I get the following _very incorrect_ output:

<img width="533" alt="image" src="https://github.com/microsoft/winget-cli/assets/12860484/434d3a45-151d-493d-96a2-717ecd1608f0">

There's a confusion between PowerShell and PowerShell Preview.

(unsure why `winget list powershell` lists oh-my-posh and Windows Terminal, but that's not the point)

### Steps to reproduce

Unsure, I haven't seen this on any other machine with the same PowerShell versions installed.

### Expected behavior

![image](https://github.com/microsoft/winget-cli/assets/12860484/fc3897ef-9884-4201-9d50-86a3cc9ebdc6)

### Actual behavior

![image](https://github.com/microsoft/winget-cli/assets/12860484/95264ac1-144a-45ee-9ccf-59bab6912a31)

### Environment

```shell
Windows Package Manager v1.6.2771
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.22621.2428
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.21.2771.0

Winget Directories
-----------------------------------------------------------------------------------------------------------------------
Logs                               %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\Diag…
User Settings                      %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\sett…
Portable Links Directory (User)    %LOCALAPPDATA%\Microsoft\WinGet\Links
Portable Links Directory (Machine) C:\Program Files\WinGet\Links
Portable Package Root (User)       %LOCALAPPDATA%\Microsoft\WinGet\Packages
Portable Package Root              C:\Program Files\WinGet\Packages
Portable Package Root (x86)        C:\Program Files (x86)\WinGet\Packages
Installer Downloads                %USERPROFILE%\Downloads

Links
---------------------------------------------------------------------------
Privacy Statement   https://aka.ms/winget-privacy
License Agreement   https://aka.ms/winget-license
Third Party Notices https://aka.ms/winget-3rdPartyNotice
Homepage            https://aka.ms/winget
Windows Store Terms https://www.microsoft.com/en-us/storedocs/terms-of-sale

Admin Setting                             State
--------------------------------------------------
LocalManifestFiles                        Disabled
BypassCertificatePinningForMicrosoftStore Disabled
InstallerHashOverride                     Disabled
LocalArchiveMalwareScanOverride           Disabled
```

**Comment by denelon** [comment]
I'm guessing some recent changes in the matching logic is including "tags" from manifests. JanDeDobbeleer.OhMyPosh has "powershell" as one of its tags.

**Comment by sba923** [comment]
> I'm guessing some recent changes in the matching logic is including "tags" from manifests. JanDeDobbeleer.OhMyPosh has "powershell" as one of its tags.

I presume this remark refers to the non-PowerShell matched packages, not to the PowerShell vs. PowerShell Preview mixup?

**Comment by denelon** [comment]
The first comment is likely the cause (unless the metadata in the registry is the same for both versions of PowerShell). The second comment is related to the non "PowerShell" packages.

**Comment by sba923** [comment]
OK.

Those (presumably recent) changes should affect all machines with both PowerShell and PowerShell Preview installed, right?

**Comment by mdanish-kh** [comment]
This is likely because `AppsAndFeaturesEntries` is missing in newer manifests which helps in package matching. I'll create PRs to add it back to see if it has an effect

**Comment by mdanish-kh** [comment]
@sba923 The PRs for adding `AppsAndFeaturesEntries` have been merged. Can you check if the issue persists?

**Comment by sba923** [comment]
I hate to say, but it does persist on one of my systems:

![image](https://github.com/microsoft/winget-cli/assets/12860484/59689913-771c-48bc-92dc-9994891bd55f)

Here are the logs, FWIW: 
[WinGet-2023-10-28.zip](https://github.com/microsoft/winget-cli/files/13194546/WinGet-2023-10-28.zip)

I will not update either PowerShell on that machine for now so that we have a system to experiment with.

**Comment by mdanish-kh** [comment]
@sba923 That installation of PowerShell doesn't look right to me :eyes: The DisplayName doesn't match any of the MSIX, MSI installations of PowerShell. They're usually `PowerShell` (msix) / `PowerShell 7-x64` (msi) etc. I couldn't repro the same on a VM: 

![image](https://github.com/microsoft/winget-cli/assets/88161975/9d2a91e7-cd48-4782-8742-4e51e42129fb)


Can you try removing the `PowerShell 7.3.8.0-x64` manually from `appwiz.cpl` and do a fresh install through `winget install Microsoft.PowerShell`?

**Comment by sba923** [comment]
Indeed, that install looks weird. Here's what the registry contains:

![image](https://github.com/microsoft/winget-cli/assets/12860484/64ddb4d8-caf7-419c-943f-bfb6fdfe6497)

And what `appwiz.cpl` shows:

![image](https://github.com/microsoft/winget-cli/assets/12860484/28275831-f80e-4914-9853-005e50955a92)

After removing PowerShell via `appwiz.wpl` I get:

![image](https://github.com/microsoft/winget-cli/assets/12860484/1a23c627-5b33-46a5-afbb-0604195b7586)

![image](https://github.com/microsoft/winget-cli/assets/12860484/677e69b7-fba0-4929-81c3-cc7b215fe8f1)

After `winget install Microsoft.PowerShell` the situation's back to normal:

![image](https://github.com/microsoft/winget-cli/assets/12860484/3ca989ad-5e7b-48af-8839-0649e4ef1373)

![image](https://github.com/microsoft/winget-cli/assets/12860484/3c81697c-1520-432d-8065-1a08c568882b)


![image](https://github.com/microsoft/winget-cli/assets/12860484/a070b066-8b4a-49b9-9222-8265f1c18e3a)

Where can that entry have come from? Update via the VScode PowerShell extension? Via Microsoft Update?

![image](https://github.com/microsoft/winget-cli/assets/12860484/c9056290-1ae5-4910-bd3d-1f3c552429f2)

**Comment by mdanish-kh** [comment]
> Where can that entry have come from? Update via the VScode PowerShell extension? Via Microsoft Update?

Honestly, I don't have a clue 😅 

> After winget install Microsoft.PowerShell the situation's back to normal:

🎉

**Comment by sba923** [comment]
> > Where can that entry have come from? Update via the VScode PowerShell extension? Via Microsoft Update?
> 
> Honestly, I don't have a clue 😅 
> 

Check out this thread:

https://twitter.com/EdTittel/status/1718294340542333148?t=E6X5pAr6KcrYkiELuT9Ejg&s=19
